### DISCUSSION TITLE
Datanode can't connect to kafka

### DISCUSSION THREAD

**Discussion by mikeolubode** [milvus-io/milvus#33574] [Q&A and General discussion] [answered]
I am running Milvus 2.3.11 and using bitnami kafka for my log broker.

Due to many critical vulnerabilities in bitnami/kafka:3.1.0-debian-10-r52, I upgraded kafka to 3.7.0-debian-12-r6
After the upgrade datanode started throwing the following errors

`FAIL|rdkafka#consumer-6| [thrd:GroupCoordinator]: GroupCoordinator: milvus-kafka-0.milvus-kafka-headless.milvus-database.svc.cluster.local:9092: Failed to │
│  resolve 'milvus-kafka-0.milvus-kafka-headless.milvus-database.svc.cluster.local:9092': Name or service not known (after 2ms in state CONNECT, 1 identical error(s) suppressed)`

I believe it is not able to connect to kafka. I ran `nslookup milvus-kafka-0.milvus-kafka-headless.milvus-database.svc.cluster.local` on a pod in the same namespace and it was able to resolve the host. But I have not idea why my datanode cannot connect to kafka.

I reverted back to the previous deployment but I keep getting the same error. I have tried restarting both kafka and datanode pods but I get the same errors.

Now, I am not sure if the error started before my kafka upgrade but I noticed it after the upgrade. 
Please help.

**Comment by yhmo** [comment, marked as answer]
I think it is a deployment problem. I didn't get a chance to try bitnami/kafka, but I think it should work if the kafka container works fine.

**Reply by mikeolubode** [reply]
It is a deployment problem. I can tell it's related to the network configurations from kafka side but still haven't been able to debug it. It's also a big update since there is 2 years between my old version and the new one.

**Reply by mikeolubode** [reply]
Update: It's not a problem. I set my error level to error so I wasn't seeing the info logs showing that the datanodes were running well. The errors were thrown while kafka was  trying to start up.
Thanks @yhmo
